### ISSUE TITLE
feat: Add rule to enforce `#{}` interpolation for SCSS functions in custom property values

### ISSUE THREAD

**Issue by lyhoang-web-dev** [stylelint-scss/stylelint-scss#1246]
## Problem

In SCSS, function calls inside custom property values are not evaluated unless wrapped in `#{}` interpolation.
No error is thrown. The bug is invisible until you inspect the computed CSS.

```scss
// ❌ SCSS does NOT evaluate this — outputs literal text in CSS
--my-color: tokens.get("color.label.fg.secondary");
//           ↑ compiles to: --my-color: tokens.get("color.label.fg.secondary")

// ✅ Correct — #{} forces SCSS to evaluate the expression first
--my-color: #{tokens.get("color.label.fg.secondary")};
```

## Existing coverage

`scss/dollar-variable-no-missing-interpolation` already handles **Sass variables** (`$foo`) in custom properties.
However, **SCSS function calls** (e.g. `namespace.fn()`, `lighten()`, `math.div()`) are not covered.

## Proposed rule

`scss/function-no-missing-interpolation-in-custom-property`

Targets declarations where the property starts with `--` and the value
contains a SCSS function call not wrapped in `#{}`.

### Violation examples

```scss
/* ❌ reported */
--color: tokens.get("color.fg");
--size: math.div(16px, 2);
--shadow: lighten($base, 10%);
```

### Accepted examples

```scss
/* ✅ accepted — properly interpolated */
--color: #{tokens.get("color.fg")};
--size: #{math.div(16px, 2)};

/* ✅ accepted — plain CSS value, no SCSS function */
--color: red;
--size: 16px;
--duration: var(--speed);
```

## Notes

- This complements `scss/dollar-variable-no-missing-interpolation` which only covers `$variable` syntax
- The detection pattern could be configurable (e.g. `{ functionPattern }`) to allow projects to specify which `namespaces/functions` to flag
